build.yaml

```diff
@@ -5611,7 +5618,7 @@ defaults:
     COREFLAGS: -fno-rtti -fno-exceptions
     CPPFLAGS: -g -Wall -Wextra -Werror -Wno-long-long -Wno-unused-parameter -DOSATOMIC_USE_INLINED=1
       -Wno-deprecated-declarations
-    LDFLAGS: -g
+    LDFLAGS: -g $(if $(subst Darwin,,$(SYSTEM)),,-framework CoreFoundation)
```

Comment:
Is this correct ? In the sense that any macos build will now have that framework in, even though they are not going to use CFStreams ? For instance, the Ruby extension.

Reply:
Or rather say: is CFStream going to be used for all MacOS stuff, not just iOS ?

Reply:
Would like MacOS stuff to be able to use iOS too. Whether to turn it on depends on `GRPC_CFSTREAM` flag.

Reply:
SGTM.